### PR TITLE
JBTM-3687 Migration of common to Jakarta

### PR DESCRIPTION
**Pull request by jmfinelli**
Initial work on [JBTM-3687](https://issues.redhat.com/browse/JBTM-3687)

JDK11 JDK17 CORE !TOMCAT !AS_TESTS !RTS !JACOCO !XTS !QA_JTA !QA_JTS_OPENJDKORB !PERF !LRA !DB_TESTS !mysql !db2 !postgres !oracle

### CODE REVIEW

**Comment by jmfinelli** [comment]
@mmusgrov, could you please double check that I have activated all needed testing axes?

**Review by mmusgrov — CHANGES_REQUESTED** [review]
This PR does not fully resolve issue JBTM-3588. If your plan is for us to fix JBTM-3588 in stages then we need to have multiple issues linked to JBTM-3588.

This PR contains a mix of whitespace clean up, pom sorting and some things related to the javax -> jakarta transition. The resolution for JBTM-3588 is such a big change that it needs to focus only on the transformation and not on miscellaneous unrelated additional code updates.

I assume this PR is preparing the way for JBTM-3588 but does not actually address the task and as such you need to a separate JBTM for it.

File: common/pom.xml [hunk 1]
```diff
@@ -5,35 +5,70 @@
...
       </resource>
     </resources>
     <plugins>
       <plugin>
-        <inherited>false</inherited>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <configuration>
+          <compilerArgs>
+            <arg>-Aorg.jboss.logging.tools.addGeneratedAnnotation=false</arg>
```
[the remark this answers is not in the document]
Comment (jmfinelli) [reply common/pom.xml:64]: > This PR does not fully resolve issue JBTM-3588. If your plan is for us to fix JBTM-3588 in stages then we need to have multiple issues linked to JBTM-3588.

Yes, I planned to use JBTM-3588 to submit multiple PRs (migrating individual modules). Your suggestion to create and use subtasks to track JBTM-3588 sounds better though. As you have a clearer picture of what modules need to be migrated to jakarta, could you please create the initial subtasks for JBTM-3588? (of course, we can create new ones as we go ahead with this work) 

> This PR contains a mix of whitespace clean up, pom sorting and some things related to the javax -> jakarta transition. The resolution for JBTM-3588 is such a big change that it needs to focus only on the transformation and not on miscellaneous unrelated additional code updates.

No problem for me, I can strip off the commits related to SortPom and raise individual PRs later (i.e. after JBTM-3588 is completed)

> I assume this PR is preparing the way for JBTM-3588 but does not actually address the task and as such you need to a separate JBTM for it.

Please, refer to my previous message about subtasks

**Comment by jmfinelli** [comment]
> This PR does not fully resolve issue JBTM-3588. If your plan is for us to fix JBTM-3588 in stages then we need to have multiple issues linked to JBTM-3588.

Yes, I planned to use JBTM-3588 to submit multiple PRs (migrating individual modules). Your suggestion to create and use subtasks to track JBTM-3588 sounds better though. As you have a clearer picture of what modules need to be migrated to jakarta, could you please create the initial subtasks for JBTM-3588? (of course, we can create new ones as we go ahead with this work) 

> This PR contains a mix of whitespace clean up, pom sorting and some things related to the javax -> jakarta transition. The resolution for JBTM-3588 is such a big change that it needs to focus only on the transformation and not on miscellaneous unrelated additional code updates.

No problem for me, I can strip off the commits related to SortPom and raise individual PRs later (i.e. after JBTM-3588 is completed)

> I assume this PR is preparing the way for JBTM-3588 but does not actually address the task and as such you need to a separate JBTM for it.

Please, refer to my previous message about subtasks

File: common/pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmfinelli) [reply common/pom.xml:64]: Already done. So far, it seems there is not any plan to move it to jakarta (the only part that should be migrated is the `Generated` annotation, which is automatically applied to generated classes -> disabling this feature [i.e. using `Generated`] removes the problem)

**Comment by jmfinelli** [comment]
I'll close this in favour of #2037

**Pull request closed without merging**